### ISSUE TITLE
The efficient way to get topdocs from int filed

### ISSUE THREAD

**Issue by raultang** [quickwit-oss/tantivy#688]
I want to get the largest number from a indexed u64 filed, but from the API seems I need first get the filed count then use RangeQuery to search, Or is there other more efficient way?
which more like equivalent to `order by <filed> desc limit 1` in sql.

```
//#1
let count = ...
let query = RangeQuery::new_u64(field, (count-1..count));
searcher.search(&query, &TopDocs::with_limit(1));
//#2
searcher.search(&AllQuery, &TopDocs::with_limit(1)); //this will scan all the docs?

//let query = RangeQuery::new_u64(field, -1); is there any implementation like this. 
```

**Comment by fulmicoton** [comment]
If it is a fast field you can do this with a `TopDocs` collector that sorts by this field. The code will slightly similar to your `#2` but not exactly. And yes it will be grossly inefficient as it will scan all docs.

For `#1`, I am not entirely sure how you get your variable `count`, but it might be possible to reduce the computational time if you have a good approximation of your lowerbound.

Granted, this is ugly too, and even slower if your lowerbound is not good.

Slightly more efficient today : you iterate through the `Searcher`s and access their `SegmentReader`. From there, you can get their `InvertedIndex` and iterate all of the terms of your field in increasing fashion using the `stream_field` method (https://tantivy-search.github.io/tantivy/tantivy/termdict/trait.TermDictionary.html#associatedtype.Streamer).

Stream the entire dictionary and keep the remaining X.

Note if your index have deletes, the terms you get are not necessarily in the index :
To be sure... You need to access the posting list associated to the term. Call `.advance()` once and check that it returns true...

In other words... It is fairly hard. 
You are the second person to ask for that. We can open a ticket to make it possible.

**Comment by raultang** [comment]
@fulmicoton Yes it is hard to maintain the lower/upper bound.
So the fast filed itself have already sorted right? I see lucene have separated  for [sort/nonsort fast values](https://lucene.apache.org/core/8_3_0/core/org/apache/lucene/index/DocValuesType.html).

**Comment by fulmicoton** [comment]
@raultang I am not sure exactly what lucene sorted fast field are... I think it means the values are sorted and encoded in an fst while the term ordinal is stored for each doc.

In your case you said you wanted that for int. You don't need them "sorted" for the AllQuery solution to work.

The value is directly bitpacked. There is no dictionary encoding in between.
